.github/workflows/golangci-lint.yaml

```
@@ -0,0 +1,26 @@
+name: golangci-lint
+on:
+  pull_request:
+    branches:
+      - main
+    types: [milestoned, opened, reopened, synchronize]
+
+permissions:
+  contents: read
+  pull-requests: read
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4.2.1
+      
+      - name: Setup golang
+        uses: ./.github/actions/golang
+      
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@971e284b6050e8a5849b72094c50ab08da042db8 #v6.1.1
```

Doubt the space matters - but for uniformity 
```suggestion
        uses: golangci/golangci-lint-action@971e284b6050e8a5849b72094c50ab08da042db8 # v6.1.1
```